Worked on steps from 1 to 4 (none of them is completed)

NahuelFedyszyn opened the pull request:
Created:
- StepOne.tsx
- StepTwo.tsx
- StepThree.tsx
- StepFour.tsx

Added Libraries:
- react-native-record-screen
- react-native-image-picker

Update:
- Private.tsx (Added routes for basic navigation)

NahuelFedyszynSiltium committed:
Worked on steps from 1 to 4 (none of them is completed)

sebasegura97 commented on src/Pages/Private/Steps/StepThree/StepThree.tsx:
```diff
@@ -0,0 +1,56 @@
+import * as ImagePicker from 'react-native-image-picker';
+import { useHistory } from 'react-router-native';
+
+
```
Esta interface Action es para usar la libreria react-native-image-picker verdad? No viene ya incluida en la misma libreria? 🤔

NahuelFedyszynSiltium committed:
Woking on steps
- Added new validation screens
- Added document_picker library
- Implemented library
- Worked on navigation logic
- Added steps_info screen

TODO:
- Show files in each validation screen
- Steps preview screen

NahuelFedyszynSiltium committed:
Added screens
- Added steps verification screen
- Added steps finish screen

sebasegura97 commented on App.tsx:
```diff
@@ -10,7 +10,7 @@ const App = () => {
   }, [user]);
 
-  if (user) {
+  if (!user) {
```
🤔  eso es para probar no mas verdad?

NahuelFedyszynSiltium committed:
Deleted unused libraries

NahuelFedyszynSiltium committed:
Merge branch 'master' into Steps

NahuelFedyszynSiltium committed:
Severla

NahuelFedyszynSiltium committed:
Several changes to adjust from master
(Previous commit message incomplete)

NahuelFedyszynSiltium committed:
Corrected buttons labels

NahuelFedyszynSiltium committed:
File collection
- Added EvidenceModel
- Added logic to steps wizard:
++ Model is initialized empty
++ Model is populated in every step
++ Finally, model needs to be uploaded to Firebase

TODO: Upload model to Firebase (Ask Seba)

NahuelFedyszynSiltium committed:
Working on display preview (validation screens)

NahuelFedyszynSiltium committed:
Trying to show bubble button on screen record

NahuelFedyszynSiltium committed:
Corrected logic

NahuelFedyszynSiltium committed:
Commented bubble button logic
(Til' this sh** works)

sebasegura97 commented on src/Pages/Private/Private.tsx:
```diff
@@ -14,6 +25,19 @@ const Private = () => {
         <Route path="/home" component={Home} />
         <Route path="/profile" component={Profile} />
         <Route path="/help" component={Help} />
+
```
Podemos mover todas las rutas de los pasos al archivo de Home.tsx? Como esta en onboarding

sebasegura97 commented on src/Pages/Private/Steps/StepFour/StepFour.tsx:
```diff
@@ -0,0 +1,82 @@
+
+  let model:EvidenceModel = location.state as EvidenceModel;
+
+  console.log(model);
```
antes de pasar a master tenemos que eliminar los logs

sebasegura97 commented on src/Pages/Private/Steps/StepFour/StepFour.tsx:
```diff
@@ -0,0 +1,82 @@
+
+  console.log(model);
+  
+  useEffect(
```
:O este use effect no tiene array de dependencias, no te esta causando muchos re-renders? Osea no tenes problemas de performance en este punto de la app?

sebasegura97 commented on src/Pages/Private/Steps/StepFour/StepFour.tsx:
```diff
@@ -0,0 +1,82 @@
+  useEffect(
+    ()=>{
+      if (response!=null){
+        console.log(response.toString());
```
delete log

sebasegura97 commented on src/Pages/Private/Steps/StepFour/StepFour.tsx:
```diff
@@ -0,0 +1,82 @@
+    model.audios = [];
+    history.push({
+      pathname:"/stepsVerification",
+      state:model,
```
Me hace un poco de ruido agregar los archivos al estado de la navegación 😢. Lo mejor seria hacerlo como un context (ejemplo en onboarding), pero si no te causa problemas de perfomance lo dejemos asi, pa no demorar mas

sebasegura97 commented on src/Pages/Private/Steps/StepsVerification/StepsVerification.tsx:
```diff
@@ -0,0 +1,68 @@
+      <Text>Audios seleccionados</Text>
+      <Box height="20px" ></Box>
+      <Box height="350px" width="100%" backgroundColor="red"></Box>
+      <Box height="30px"></Box>
```
Estos box con height='30px' son para separar el contenido? En ese caso, no seria mejor usar margin='30px'?
